flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartitionProvider.java

```diff
@@ -29,4 +31,28 @@ ResultSubpartitionView createSubpartitionView(
             int index,
             BufferAvailabilityListener availabilityListener)
             throws IOException;
+
+    /**
+     * If the upstream task's partition has been registered, returns the result subpartition input
+     * view immediately, otherwise save the notifier and return null.
+     *
+     * @param partitionId the result partition id
+     * @param index the index
+     * @param availabilityListener the buffer availability listener
+     * @param notifier the partition request notifier
+     * @return the result subpartition view
+     * @throws IOException the thrown exception
+     */
+    ResultSubpartitionView createSubpartitionViewOrNotify(
+            ResultPartitionID partitionId,
+            int index,
+            BufferAvailabilityListener availabilityListener,
+            PartitionRequestNotifier notifier) throws IOException;
```

Comment:
Why does't it return `ResultSubpartitionView`? I think it would be simpler if it returned `boolean` true/false, whether the request has been processed immediately. Or even return `void` , as this is not used in production code, and maybe unit test can make an assertion in another way?